sklearn/ensemble/voting_classifier.py

```diff
@@ -163,6 +175,10 @@ def fit(self, X, y, sample_weight=None):
         if n_isnone == len(self.estimators):
             raise ValueError('All estimators are None. At least one is '
                              'required to be a classifier!')
+
+        if not self.flatten_transform and self.voting is 'soft':
```

Comment:
This condition means the user who wants flatten_transform=False will always get the warning. We can avoid this by defaulting to some special marker, eg 'default'. We can tell the user in the warning message that setting it explicitly will Silence the warning